src/lib/reasoners/fun_sat.ml

```diff
@@ -1656,6 +1669,12 @@ module Make (Th : Theory.S) = struct
           in
           acc.model_gen_phase := false;
           env.last_saved_model := None;
+          let () =
+            try env.declare_top := Stack.pop env.declare_tail
+            with Stack.Empty ->
+              (* Happens if we perform more pops than pushes. *)
+              invalid_arg "Fun_sat.pop"
```

Comment:
There is no mechanism to catch it if you use the SAT API with the library. But there is a mechanism to prevent it with the binary and this hero is Dolmen :)

I may create a new error in `Error` to manage this case?

Reply:
An exception is appropriate for library use, thanks for checking!